packages/node_modules/@webex/plugin-meetings/src/members/index.js

```diff
@@ -28,66 +29,239 @@ export default class Members extends StatelessSparkPlugin {
+        }
+      }
+      this.hostId = newHostId;
+      this.trigger(EVENT_TRIGGERS.MEMBERS_HOST_UPDATE, {
+        activeHostId: newHostId,
+        endedHostId: oldHostId
+      });
+    });
+  }
+
+  /**
+   * set up the locus info participants update listener
+   * update members collection value for members
+   * notifies consumer with members:update
+   * @returns {undefined}
+   * @event {members:update}
+   * @memberof Members
+   */
+  setUpLocusParticipantsListener() {
+    Events.on(EVENTS.LOCUS_INFO_UPDATE_PARTICIPANTS, (payload) => {
+      this.handleParticipantsPayload(payload);
     });
   }
 
   /**
+   * when new participant updates come in, both delta and full participants, update them in members collection
+   * delta object in the event will have {updated, added} and full will be the full membersCollection
    * @param {Object} payload
-   * @returns {Object}
+   * @event members:update {delta, full}
+   * @returns {undefined}
+   * @memberof Members
    */
-  handleLocusInfoUpdatedParticipants(payload) {
-    const membersUpdate = this.membersCollection.update(payload);
+  handleParticipantsPayload(payload) {
+    if (payload) {
+      const delta = this.handleLocusInfoUpdatedParticipants(payload);
+      const full = this.handleMembersUpdate(delta); // SDK should propagate the full list for both delta and non delta updates
+      this.trigger(EVENT_TRIGGERS.MEMBERS_UPDATE, {delta, full});
+    }
+  }
+
+  /**
+   * sets values in the members collection for updated and added properties from delta
+   * @param {Object} membersUpdate {updated: [], added: []}
+   * @returns {Object} membersCollection
+   * @memberof Members
+   */
+  handleMembersUpdate(membersUpdate) {
     if (membersUpdate) {
-      this.trigger('MEMBERS_UPDATE', membersUpdate);
+      if (membersUpdate.updated) {
+        this.constructMembers(membersUpdate.updated);
+      }
+      if (membersUpdate.added) {
+        this.constructMembers(membersUpdate.added);
+      }
     }
-    return membersUpdate;
+    return this.membersCollection.getAll();
+  }
+
+  /**
+   * set members to the member collection from each updated/added lists as passed in
+   * @param {Array} list
+   * @returns {undefined}
+   * @memberof Members
+   */
+  constructMembers(list) {
+    list.forEach((member) => {
+      this.membersCollection.set(member.id, member);
+    });
+  }
+
+  /**
+   * set up the locus info media shares listener
+   * update content sharing id value for members, and updates the member
+   * notifies consumer with members:content:update {activeContentSharingId, endedContentSharingId}
+   * @returns {undefined}
+   * @event {members:content:update}
+   * @memberof Members
+   */
+  setUpLocusMediaSharesListener() {
+    Events.on(EVENTS.LOCUS_INFO_UPDATE_MEDIA_SHARES, (payload) => {
+      const currentContent = payload.current;
+      let whoSharing = null;
+      let whoStopped = null;
+      if (currentContent && currentContent.contentId) {
```

Comment:
Do we handle the scenario where the user is new and started sharing at the same time.
We can do  const shareMember = this.membersCollection.get(whoSharing); on line 209 
Check if exists then use that in line 212 and 215

Reply:
it does handle that case from testing